lib/theme_check/html_node.rb

```diff
@@ -27,23 +90,39 @@ def value
     def children
       @children ||= @value
         .children
-        .map { |child| HtmlNode.new(child, theme_file, @placeholder_values, self) }
+        .map { |child| HtmlNode.new(child, theme_file, @placeholder_values, @parseable_source, self) }
     end
 
     def markup
-      @markup ||= replace_placeholders(@value.to_html)
+      @markup ||= replace_placeholders(parseable_markup)
```

Comment:
Can't use @value.to_html because of the following:
```ruby
HtmlNode.parse(<<~HTML).to_html == <<~EXPECTED
  <script 
    src="hello" 
    defer
  >
  </script>
HTML
  <script src="hello" defer=""></script>
EXPECTED
```

Basically, to_html is _built_. It's not from the source.